docs/06_how-to-guides/40_multi-index/how-to-iterate-and-retrieve-a-multi_index-table-based-on-secondary-index.md

```diff
@@ -1,72 +1,51 @@
-         testtab(receiver, receiver.value)
-         { }
+* The [`multi-index`](../../classeosio_1_1multi__index) class.
+* The [`multi-index::find(...)`](../../group__multiindex#function-find) method.
 
-      // the row structure of the multi index table, that is, each row of the table
-      // will contain an instance of this type of structure
-      struct [[eosio::table]] test_table {
-        // this field stores a name for each row of the multi index table
-        name test_primary;
-        name secondary;
-        // additional data stored in table row
-        uint64_t datum;
-        // mandatory definition for primary key getter
-        uint64_t primary_key( ) const { return test_primary.value; }
-        uint64_t by_secondary( ) const { return secondary.value; }
-      };
+## Before you begin
 
-      // the multi index type definition, for ease of use define a type alias `test_tables`, 
-      // based on the multi_index template type, parametarized with a random name, the 
-      // test_table data structure, and the secondary index
-      typedef eosio::multi_index<"testtaba"_n, test_table, eosio::indexed_by<"secid"_n, eosio::const_mem_fun<test_table, uint64_t, &test_table::by_secondary>>> test_tables;
+Make sure you have the following prerequisites in place:
 
-      // the multi index table instance declared as a data member of type test_tables
-      test_tables testtab;
+* An EOSIO development environment, for details consult the [Get Started Guide](https://developers.eos.io/welcome/latest/getting-started-guide/index),
+* A multi-index `testab` table instance which stores `user` objects indexed by the primary key which is of type `eosio::name` and a secondary index for data member `secondary` of type `eosio::name` accessible through `by_secondary()` method. Consult the section [How to define a secondary index](./how-to-define-a-secondary-index) to learn how to set it up.
 
-      [[eosio::action]] void set( name user );
-      [[eosio::action]] void print( name user );
+## Procedure
 
-      using set_action = action_wrapper<"set"_n, &multi_index_example::set>;
-      using print_action = action_wrapper<"print"_n, &multi_index_example::print>;
-};
-```
+Complete the following steps to iterate, retrieve and print data from the `testtab` multi-index table using the secondary index.
 
-To iterate and retreive the multi index table `testtab` defined in `multi_index_example` contract based on secondary index `by_secondary`, define a third action `bysec` which will do exactly that.
+### 1. Define The bysec(...) Action
 
-1. In the contract definition, add the new action definition, using the `[[eosio::action]] void` and the `eosio::action_wrapper` template like this:
+Add to the definition of the multi-index table the `bysec` action which gets as parameter an account name. This will be the action which will retrieve the user object stored in the multi-index based on the name input parameter using the secondary index.
```

Comment:
Maybe:

Add an action to the definition of the multi-index table which takes the account name as a parameter. This action will retrieve the user object stored in the multi-index based on the passed in account name parameter. The action will use the secondary index.